affordance_primitives/include/affordance_primitives/task_estimator/task_estimator.hpp

```diff
@@ -57,8 +57,9 @@ class TaskEstimator
   /** Resets the internal estimation
    *
    * @param reset_val The value to reset to
+   * @return True if successful, false otherwise
    */
-  virtual void resetTaskEstimation(double reset_val = 0) = 0;
+  virtual bool resetTaskEstimation(double reset_val = 0) = 0;
 
   virtual ~TaskEstimator(){};
```

Comment:
I might have missed the constructor in this file, but is there a reason to declare a destructor without a constructor? 
I get the virtual destructor helps to avoid undefined behavior when deleting a derived class object using a pointer of base class type. However, is there a specific behavior we are destructing here? And if so, shouldn't we have a constructor as well? And if not, are we okay with a default constructor in this case?

Reply:
Most this came from me following the [pluginlib tutorial](http://wiki.ros.org/pluginlib/Tutorials/Writing%20and%20Using%20a%20Simple%20Plugin), however, this no-constructor but virtual public destructor is good for a pure virtual base class in C++, [see here](https://stackoverflow.com/questions/40666985/destructor-and-no-constructor-for-abstract-base-class-c) for details.